File: src/main/java/org/veupathdb/lib/blast/field/LineLength.java
package org.veupathdb.lib.blast.field;

import com.fasterxml.jackson.databind.JsonNode;
import org.veupathdb.lib.blast.util.DefaultingJSONValue;
import org.veupathdb.lib.blast.util.JSONConstructor;

public record LineLength(int value) implements DefaultingJSONValue
{
  public static final int DefaultValue = 60;

  @Override
  public boolean isDefault() {
    return value == DefaultValue;
  }

  @Override
  public JsonNode toJSON() {
    return JSONConstructor.newInt(value);
  }
}

File: src/main/java/org/veupathdb/lib/blast/field/LineLength.kt
package org.veupathdb.lib.blast.field

import com.fasterxml.jackson.databind.JsonNode
import kotlin.jvm.JvmInline
import org.veupathdb.lib.blast.util.DefaultingJSONValue
import org.veupathdb.lib.blast.util.JSONConstructor

@JvmInline
value class LineLength(val value: Int) : DefaultingJSONValue
{
  companion object {
    const val DefaultValue = 60
  }

  override val isDefault: Boolean
    get() = value == DefaultValue

  override fun toJSON(): JsonNode = JSONConstructor.newInt(value)
}
